controllers/btpoperator_controller.go

```diff
@@ -417,11 +434,11 @@ func (r *BtpOperatorReconciler) getResourcesToApplyPath() string {
 	return fmt.Sprintf("%s%capply", ResourcesPath, os.PathSeparator)
 }
 
-func (r *BtpOperatorReconciler) prepareModuleResources(ctx context.Context, us []*unstructured.Unstructured, s *corev1.Secret) error {
+func (r *BtpOperatorReconciler) prepareModuleResourcesFromManifests(ctx context.Context, resourcesToApply []*unstructured.Unstructured, s *corev1.Secret) error {
 	logger := log.FromContext(ctx)
 
 	var configMapIndex, secretIndex int
-	for i, u := range us {
+	for i, u := range resourcesToApply {
```

Comment:
Is is possible that we won't find secret in the resources? How we should handle this?
The same for configMap. We should decide whether we should handle or ignore (leave as it is).

Reply:
Both, Secret and ConfigMap, are part of the module resources atm, but we can't be sure whether they are here to stay. We should handle the case when they are missing, throw an error and fail fast since they are required for the module to work.

Reply:
I will create a separate PR with fix